huaweicloud/services/acceptance/as/data_source_huaweicloud_as_policies_test.go

```diff
@@ -0,0 +1,184 @@
+package as
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/common"
+)
+
+func TestAccPoliciesDataSource_basic(t *testing.T) {
+	var (
+		dataSourceName = "data.huaweicloud_as_policies.test"
+		dc             = acceptance.InitDataSourceCheck(dataSourceName)
+		baseConfig     = testACCPolicy_base()
+	)
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck: func() {
+			acceptance.TestAccPreCheck(t)
+		},
+		ProviderFactories: acceptance.TestAccProviderFactories,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccPoliciesDataSource_basic(baseConfig),
+				Check: resource.ComposeTestCheckFunc(
+					dc.CheckResourceExists(),
+					resource.TestCheckResourceAttrSet(dataSourceName, "policies.0.scaling_group_id"),
+					resource.TestCheckResourceAttrSet(dataSourceName, "policies.0.id"),
+					resource.TestCheckResourceAttrSet(dataSourceName, "policies.0.name"),
+					resource.TestCheckResourceAttrSet(dataSourceName, "policies.0.status"),
+					resource.TestCheckResourceAttrSet(dataSourceName, "policies.0.type"),
+					resource.TestCheckResourceAttrSet(dataSourceName, "policies.0.alarm_id"),
+					resource.TestCheckResourceAttrSet(dataSourceName, "policies.0.action.0.operation"),
+					resource.TestCheckResourceAttrSet(dataSourceName, "policies.0.action.0.instance_number"),
+					resource.TestCheckResourceAttrSet(dataSourceName, "policies.0.cool_down_time"),
+					resource.TestCheckResourceAttrSet(dataSourceName, "policies.0.created_at"),
+					resource.TestCheckOutput("is_scaling_policy_id_filter_useful", "true"),
+					resource.TestCheckOutput("is_scaling_policy_name_filter_useful", "true"),
+					resource.TestCheckOutput("is_scaling_policy_type_filter_useful", "true"),
+				),
+			},
+		},
+	})
+}
+
+//nolint:revive
+func testACCASGroup_base(rName string) string {
+	return fmt.Sprintf(`
+%[1]s
+
+resource "huaweicloud_compute_keypair" "acc_key" {
```

Comment:
huaweicloud_kps_keypair is prefer, and we don't need to import a public key.

Reply:
fixed.